### PR TITLE
Bump rpush-redis and remove hiredis dependency

### PR DESCRIPTION
**Pull request by fdoxyz**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description

The new releases of `rpush-redis` and `modis` gems now don't include `hiredis` as a dependency.

## Related Tickets & Documents

N/A

## QA Instructions, Screenshots, Recordings

Test suite coverage should be able to validate we're covered here. You could also try to connect to a TLS-only Redis to ensure `hiredis` is no longer taking control over connections (and failing due to TLS not being supported by them).

### UI accessibility concerns?

None - Backend only

## Added tests?

- [x] No, and this is why: It's only a dependency bump and existing test coverage should be enough

## [Forem core team only] How will this change be communicated?

- [x] I will share this change internally with the appropriate teams

## [optional] Are there any post deployment tasks we need to perform?

Keeping an eye on Redis connections to ensure nothing breaks when `hiredis` is no longer used would be good

## [optional] What gif best describes this PR or how it makes you feel?

safety

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -352,7 +352,6 @@ GEM
     hashie (4.1.0)
     heapy (0.2.0)
       thor
-    hiredis (0.6.3)
```
Comment (rhymes) [inline Gemfile.lock:355]: bye bye bye

**Review by rhymes — APPROVED** [review]
YES!

**Review by jgaskins — APPROVED** [review]
THANK YOU

**Review by juliannatetreault — APPROVED** [review]
🚀 !

**Pull request merged**